### PR TITLE
Allow to change forge status messages

### PR DESCRIPTION
**Pull request by qwerty287**
Allow to change the status message via template option

Closes https://github.com/woodpecker-ci/woodpecker/issues/855

### CODE REVIEW

**Comment by confusedsushi** [comment]
I like the current context text. From a programmers point of view it is short, informative and probably easy to parse.

I see that someone maybe wants a more human friendly text. What about making the format string which builds the text configurable?

File: server/remote/common/status.go [hunk 1]
```diff
@@ -23,10 +23,10 @@ func GetBuildStatusContext(repo *model.Repo, build *model.Build, proc *model.Pro
 		name += "/" + proc.Name
 	}
 
-	return name
+	return fmt.Sprintf("%s - %s (%s)", server.Config.Server.StatusContext, proc.Name, build.Event)
```
Comment (6543) [inline server/remote/common/status.go:26]: will make the other logic of GetBuildStatusContext useless & is breaking

**Comment by 6543** [comment]
well just let it ope until we could run down the 2 points :)

File: cmd/server/flags.go [hunk 2]
```diff
@@ -206,6 +206,12 @@ var flags = []cli.Flag{
 		Usage:   "status context prefix",
 		Value:   "ci/woodpecker",
 	},
+	&cli.StringFlag{
+		EnvVars: []string{"WOODPECKER_STATUS_CONTEXT_FORMAT", "WOODPECKER_GITHUB_CONTEXT_FORMAT", "WOODPECKER_GITEA_CONTEXT_FORMAT"},
```
Comment (6543) [inline cmd/server/flags.go:210]:
```suggestion
		EnvVars: []string{"WOODPECKER_STATUS_CONTEXT_FORMAT"},
```
the gitea/github env var above where just keeped, to not break too much

**Comment by 6543** [comment]
adding tests is always a good idea ;)

**Review by 6543 — APPROVED** [review]

File: docs/docs/30-administration/10-server-config.md [hunk 3]
```diff
@@ -305,6 +305,17 @@ Read the value for `WOODPECKER_PROMETHEUS_AUTH_TOKEN` from the specified filepat
 
 Context prefix Woodpecker will use to publish status messages to SCM. You probably will only need to change it if you run multiple Woodpecker instances for a single repository.
 
+### `WOODPECKER_STATUS_CONTEXT_FORMAT`
+> Default: `%context/%event/%pipeline`
+
+Template for the status messages published to SCM.
```
Comment (anbraten) [inline docs/docs/30-administration/10-server-config.md:311]:
```suggestion
Template for the status messages published to forges.
```

File: server/remote/common/status.go [hunk 4]
```diff
@@ -2,31 +2,32 @@ package common
...
+	"strings"
 
 	"github.com/woodpecker-ci/woodpecker/server"
 	"github.com/woodpecker-ci/woodpecker/server/model"
 )
 
 func GetBuildStatusContext(repo *model.Repo, build *model.Build, proc *model.Proc) string {
-	name := server.Config.Server.StatusContext
-
+	ctx := server.Config.Server.StatusContextFormat
```
Comment (anbraten) [inline server/remote/common/status.go:12]: What do you think about using GOs template function instead?

Comment (anbraten) [reply server/remote/common/status.go:12]: https://pkg.go.dev/text/template

Comment (jolheiser) [reply server/remote/common/status.go:12]: Either text template or https://pkg.go.dev/os#Expand style imo

Comment (6543) [reply server/remote/common/status.go:12]: @qwerty287 u may want to look into this ;) ?

**Comment by 6543** [comment]
please update the unittests too https://ci.woodpecker-ci.org/woodpecker-ci/woodpecker/build/2472/31

**Review by 6543 — APPROVED** [review]

**Pull request merged**